apps/src/applab/RecordListener.js

```diff
@@ -0,0 +1,230 @@
+'use strict';
+
+/**
+ * Class which allows callers to listen to changes to data tables. This is currently
```

Comment:
I've listed remaining work items required to publicly enable and advertise this block in the description of https://github.com/code-dot-org/code-dot-org/pull/6915